switchlink/sai/switchlink_handle_lag.c

```diff
@@ -0,0 +1,383 @@
+  sai_status_t status = SAI_STATUS_SUCCESS;
+
+  if (lag_intf->link_type == SWITCHLINK_LINK_TYPE_BOND) {
+    sai_attribute_t attr_list[5];
+    int ac = 0;
+    memset(attr_list, 0, sizeof(attr_list));
+    attr_list[ac].id = SAI_LAG_ATTR_CUSTOM_RANGE_START;
+    memcpy(attr_list[ac].value.mac, lag_intf->mac_addr, sizeof(sai_mac_t));
+    ac++;
+
+    attr_list[ac].id = SAI_LAG_ATTR_INGRESS_ACL;
+    attr_list[ac].value.oid = 0;
+    if (!(lag_h)) {
+      krnlmon_log_error("LAG handle is NULL");
+      return -1;
+    } else {
+      attr_list[ac].value.oid = *lag_h;
+    }
+    ac++;
+    status = sai_lag_api->create_lag(lag_h, 0, ac, attr_list);
+  }
+  return ((status == SAI_STATUS_SUCCESS) ? 0 : -1);
+}
+
+/*
+ * Routine Description:
+ *    SAI call to delete lag
+ *
+ * Arguments:
+ *    [in] lag - lag info
+ *    [in] lag_h - lag handle
+ *
+ * Return Values:
+ *    0 on success
+ *   -1 in case of error
+ */
+static int delete_lag(const switchlink_db_interface_info_t* lag,
+                      switchlink_handle_t lag_h) {
+  sai_status_t status = SAI_STATUS_SUCCESS;
+  if (lag->link_type == SWITCHLINK_LINK_TYPE_BOND) {
+    status = sai_lag_api->remove_lag(lag_h);
+  }
+  return ((status == SAI_STATUS_SUCCESS) ? 0 : -1);
+}
+
+/*
+ * Routine Description:
+ *    SAI call to set lag attribute
+ *
+ * Arguments:
+ *    [in] lag - lag info
+ *    [out] lag_h - lag member handle
+ *
+ * Return Values:
+ *    0 on success
+ *   -1 in case of error
+ */
+static int set_lag_attribute(const switchlink_db_interface_info_t* lag_info,
+                             switchlink_handle_t lag_member_h) {
+  sai_attribute_t attr;
+  memset(&attr, 0, sizeof(attr));
+
+  attr.id = SAI_LAG_ATTR_EGRESS_ACL;
+  attr.value.oid = lag_member_h;
+  return sai_lag_api->set_lag_attribute(lag_info->lag_h, &attr);
+}
+
+/*
+ * Routine Description:
+ *    SAI call to create lag member
+ *
+ * Arguments:
+ *    [in] lag - lag member info
+ *    [out] lag_member_h - lag member handle
+ *
+ * Return Values:
+ *    0 on success
+ *   -1 in case of error
+ */
+static int create_lag_member(
+    const switchlink_db_lag_member_info_t* lag_member_intf,
+    switchlink_handle_t* lag_member_h) {
+  sai_attribute_t attr_list[5];
+  int ac = 0;
+  bf_status_t bf_status;
+  uint32_t port_id = 0;
+  bf_dev_id_t bf_dev_id = 0;
+  static char mac_str[SWITCH_PD_MAC_STR_LENGTH];
+
+  snprintf(mac_str, sizeof(mac_str), "%02x:%02x:%02x:%02x:%02x:%02x",
+           lag_member_intf->perm_hwaddr[0], lag_member_intf->perm_hwaddr[1],
+           lag_member_intf->perm_hwaddr[2], lag_member_intf->perm_hwaddr[3],
+           lag_member_intf->perm_hwaddr[4], lag_member_intf->perm_hwaddr[5]);
+  mac_str[SWITCH_PD_MAC_STR_LENGTH - 1] = '\0';
+
+  bf_status = bf_pal_get_port_id_from_mac(bf_dev_id, mac_str, &port_id);
+  if (bf_status != BF_SUCCESS) {
+    port_id = lag_member_intf->perm_hwaddr[1] + SWITCH_PD_TARGET_VPORT_OFFSET;
+    krnlmon_log_debug(
```

Comment:
use krnlmon_log_info instead of debug here, also modify the same in "switch_pd_utils.c"  file as well

Reply:
Addressed.